app/controllers/state_file/questions/submission_confirmation_controller.rb

```diff
@@ -1,7 +1,9 @@
 module StateFile
   module Questions
     class SubmissionConfirmationController < QuestionsController
-      def edit; end
+      def edit
+        raise ActiveRecord::RecordNotFound unless EfileSubmission.where(data_source: current_intake).present?
```

Comment:
noticed we would receive a weird error on this page if we generated a intake from the flow service and then immediately went to the submission page. This is because we need a submission on this page and we can only get one from submitting on the esign_declaration form. Hopefully this error message makes it more obvious in those cases.